js/base/Exchange.js

```diff
@@ -2541,7 +2560,9 @@ module.exports = class Exchange {
             if (symbol in this.markets) {
                 return this.markets[symbol];
             } else if (symbol in this.markets_by_id) {
-                return this.markets_by_id[symbol];
+                // we insert spot markets first so this will return a spot market
+                // if there is a conflict between the spot and swap markets
+                return this.markets_by_id[symbol][0];
```

Comment:
DETECT LANGUAGE

ENGLISH

PORTUGUESE

RUSSIAN

PORTUGUESE

ENGLISH

SPANISH
Source text
this is a breaking change, some users use the exchange-specific swap ids so their code will break because now we will be returning a spot market instead.  I think we should avoid it